pkg/networkmanager/firewall-client.js

```diff
@@ -146,7 +146,9 @@ function fetchServiceInfos(services) {
      * Work around `cockpit.all()` returning results in individual arguments -
      * that's just confusing and doesn't work with ES6 style functions.
      */
-    return promises.then(function () {
+    return promises.then(function (services) {
+        if (Array.isArray(services) && services.length === 0)
+            return [];
         return Array.prototype.slice.call(arguments);
```

Comment:
this is very confusing but correct. I'll add a comment. Services is badly named here, it's the first item in the list of services; and if that first argument is empty, then this function was called on an empty array.

The `slice.call` will otherwise return `[undefined]`